pinot-core/src/main/java/com/linkedin/pinot/core/operator/query/MetadataBasedAggregationOperator.java

```diff
@@ -46,23 +49,17 @@
    *
    * @param aggregationFunctionContexts Aggregation function contexts.
    * @param segmentMetadata Segment metadata.
-   * @param dataSourceMap Map of column to its data source.
+   * @param dictionaryMap Map of column to its dictionary.
    */
   public MetadataBasedAggregationOperator(AggregationFunctionContext[] aggregationFunctionContexts,
-      SegmentMetadata segmentMetadata, Map<String, BaseOperator> dataSourceMap) {
+      SegmentMetadata segmentMetadata, Map<String, Dictionary> dictionaryMap) {
```

Comment:
This operator is based on Dictionary and it probably makes sense to have DictionaryBasedAggregationOperator instead of overloading MetadataBasedAggregationOperator